File: newton/newton.py
# -*- coding: utf-8 -*-
# Newton
#La función a evaluar es f(x)=0.5p^(-0.2)+0.5p^(-0.5)-2=0
import sys
#Parametros
diff=1
tol=1.0e-6
contador=0
#Funciones
f = lambda x: 0.5*x**(-0.2)+0.5*x**(-0.5)-2
df = lambda x: -0.1*x**(-1.2)-0.25*x**(-1.5)
#Punto inicial
x0=0.1

#Busqueda    
while diff > tol:
    f0=f(x0)
    df0=df(x0)
    x=x0-f0/df0
    diff=abs(x-x0)
    x0=x
    contador = contador +1
    print 'iteración: %d, distancia: %0.7f' % (contador, diff)
    if contador >= 200:
        print 'Error: Maximo numero de iteraciones alcanzado'
        sys.exit()
fx=f(x)
print 'La solución es x: %0.7f, y el valor de la función es fx=: %0.9f' % (x, fx)
#print 'Solución: '+str(x)

File: newton/newton.f90
! PROGRAM newton

program newton

    ! variable declaration
    implicit none
    integer :: contador
    real*8 :: x, xo, fx, diff, tol, fxo, dfxo
    

    ! Parametros
    diff=1
    tol=1d-6
    contador=0

    ! Punto inicial
    xo=0.1d0
    

    ! Busqueda
    do while (diff > tol)

      fxo=testfun(xo)
      dfxo=testdif(xo)
      x=xo-fxo/dfxo
      diff=abs(x-xo)
      xo=x
      contador=contador +1

      if (contador > 200)then
        write(*,'(/a)')' Numero maximo de iteraciones alcanzado '
        stop
      endif
      

      write(*,'(/a,i4,a,f12.7)')' Iteracion  ',contador,'  diferencia ',diff,
      
    enddo

    fx=testfun(x)

    write(*,'(/a,f12.7,a,f12.9)')' x = ',x,'    f = ',fx,


    contains

    function testfun(x)

        implicit none
    
        ! input arguments
        real*8, intent(in) :: x
    
        ! function value
        real*8 :: testfun
          
        ! executable code
        testfun = 0.5d0*x**(-0.2d0)+0.5d0*x**(-0.5d0)-2d0
    
    end function

    function testdif(x)

        implicit none
    
        ! input arguments
        real*8, intent(in) :: x
    
        ! function value
        real*8 :: testdif
          
        ! executable code
        testdif = -0.1d0*x**(-1.2d0)-0.25d0*x**(-1.5d0)
    
    end function


end program